lib/clients/statsd.js

```diff
@@ -0,0 +1,79 @@
+var BaseClient = require('./base');
+var util = require('util');
+var types = require('../types');
+var _ = require('lodash');
+var dgram = require('dgram');
+
+var STATS_TYPES = {};
+STATS_TYPES[types.C] = 'c';
+STATS_TYPES[types.G] = 'g';
+STATS_TYPES[types.T] = 'ms';
+
+var KEY_DELIMITER = "-";
+
+function sanatizeInput(input) {
+  //Replace characters that may cause problems here (:| are used by statsd, and cannot be included in the name)
+  return (input + '').replace(/[\s:|=<.]+/g, '');
+}
+
+/**
+ * A client that can send metrics data to a statsd backend
+ * @param {Object} opts options about the statsd backend
+ * @param {String} opts.host the host of the statsd server. Default is 127.0.0.1.
+ * @param {Number} opts.port the port of the statsd server. Default is 8125.
+ */
+function StatsdClient(opts) {
+  BaseClient.apply(this, arguments);
+  opts = opts || {};
+  this.host = opts.host || '127.0.0.1';
+  this.port = opts.port || 8125;
+  this.socket = dgram.createSocket('udp4');
+}
+
+util.inherits(StatsdClient, BaseClient);
+
+//statsd only allows one value per message. If the data contains multiple fields, each of the field will be converted to a message
+//also it doesn't support tags. So we will embed tags into the key. Each message will be constructed like this:
+// <key>-<tagKey>_is_<tagValue>...-<fieldName>:<fieldValue>|<type>
+//For example, given this data:
+// {type: 'gauge', key: 'memory', tags: {host: localhost, workerId:1}, fields: {used: 800, total: 1000}}
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)